### DISCUSSION TITLE
CC-Tweaked on Bedrock

### DISCUSSION THREAD

**Discussion by FrederoxDev** [cc-tweaked/CC-Tweaked#1910] [Show and tell]
Hey my name is FrederoxDev and I am a mod developer for MCBE. When I say mod developer, for clarity, I am referring to actual native mods to the game and not addons.

I found this [other issue](https://github.com/cc-tweaked/CC-Tweaked/issues/1353), if the technology was there, it seems like the idea of bedrock is pretty open. I run a modding platform, similar to forge/fabric but on bedrock called [Amethyst](https://github.com/FrederoxDev/Amethyst).

Recently I have been experimenting with implementing the mod and would like to eventually publish it. Its quite barebones right now, but thought I should ask earlier rather than later. Bedrock modding is quite different to java edition and on the client we have no form of "mappings" like java edition does, so everything that I am using here, I have had to reverse engineer and identify on the client.

The white box in this video represents a turtle (you've got to use your imagination here!)

https://github.com/cc-tweaked/CC-Tweaked/assets/69014593/3c1551d4-811b-4195-8e97-9efff194b215

The code for the project is hosted here: https://github.com/FrederoxDev/CC-Amethyst although I imagine if you are not familiar with this type of modding, some parts might be a bit hard to read. (And plus its my first real time doing anything with multithreading, so it might not be the best code lol)

**Comment by CelDaemon** [comment]
With the recent removal of bedrock PDBs, will this still be feasible?

**Comment by FrederoxDev** [comment]
I personally just plan to stick on the version 1.21.0.3 similar to what developers have done on Java for some versions. 

It's technically possible to forward port too but it would take a lot more effort
